src/compiler/path.ts

```diff
@@ -756,6 +757,24 @@ export function changeAnyExtension(path: string, ext: string, extensions?: strin
     return pathext ? path.slice(0, path.length - pathext.length) + (startsWith(ext, ".") ? ext : "." + ext) : path;
 }
 
+/**
+ * @internal
+ * Like `changeAnyExtension`, but declaration file extensions are recognized
+ * and replaced starting from the `.d`.
+ *
+ * ```ts
+ * changeAnyExtension("file.d.ts", ".js") === "file.d.js"
+ * changeFullExtension("file.d.ts", ".js") === "file.js"
+ * ```
+ */
+export function changeFullExtension(path: string, newExtension: string) {
+    const declarationExtension = getDeclarationFileExtension(path);
+    if (declarationExtension) {
+        return path.slice(0, path.length - declarationExtension.length) + newExtension;

```

Comment:
Does this need to handle presence or absence of `"."` in new extension like in `changeAnyExtension` : `(startsWith(ext, ".") ? ext : "." + ext)`